test/ts-types/main.ts

```diff
@@ -0,0 +1,27 @@
+import {getDocument} from "pdfjs-dist";
+
+class MainTest {
+    task: ReturnType<typeof getDocument> | undefined
+
+    constructor(public file: string) {
+    }
+
+    loadPdf() {
+        this.task = getDocument("file://" + this.file);
+        return this.task.promise;
+    }
+
+    get pdf() {
+        return this.task?.promise;
+    }
+
+    get page() {
+        return this.pdf?.then(doc => doc.getPage(1));
+    }
+
+}
+
+const mt = new MainTest("./test.pdf");
```

Comment:
There is no such file in this directory, so how can this work? I presume this `main.ts` file is used to ensure that the generated types can be used again, which is great, so can't this test be even more minimal by simply using some of the types and being done? No need to actually load a file I think.

Reply:
In fact the test has only one goal: check if it can compile `main.ts` with `strict: true` (which is even more strictly checking everything compiles well). Once the compilation succeeds, the test is successful.

Personally I would've liked to also import some other types, but @tamuratak insisted that only one method is imported here. Seeing that it correctly failed for the missing `annotationStorage` parameter, this seems to be enough.

Reply:
Yes, that makes the intent more clear.